### PR TITLE
security(superuser): Log superuser grants

### PR DESCRIPTION
**Pull request by Milkshak3s**
Logs "Reason" and "Access Category" for successful superuser grants via the superuser modal.

### CODE REVIEW

**Comment by Milkshak3s** [comment]
This was tested locally, and produced expected logs without error.

**Review by misha-sentry — APPROVED** [review]

**Comment by misha-sentry** [comment]
@Milkshak3s were you able to test locally (if that's possible for this) and is there a screenshot or anything you can share for that?

File: src/sentry/api/endpoints/auth_index.py [hunk 1]
```diff
@@ -257,6 +257,15 @@ def put(self, request: Request):
 
         if request.user.is_superuser and request.data.get("isSuperuserModal"):
             request.superuser.set_logged_in(request.user)
+            logging.info(
```
Comment (dashed) [inline src/sentry/api/endpoints/auth_index.py:260]: @Milkshak3s Would it not be better to move this logging into the `request.superuser.set_logged_in()` function in https://github.com/getsentry/sentry/blob/ea9e2e29653c5eb60741d68a1220657e24609eeb/src/sentry/auth/superuser.py#L328?

**Review by dashed — CHANGES_REQUESTED** [review]

File: src/sentry/api/endpoints/auth_index.py (hunk 1 above)
Comment (Milkshak3s) [reply src/sentry/api/endpoints/auth_index.py:260]: Ah! Looks like this logging case is already covered in that function:

https://github.com/getsentry/sentry/blob/ea9e2e29653c5eb60741d68a1220657e24609eeb/src/sentry/auth/superuser.py#L387-L394

I'll close this out. Thanks @dashed !

**Comment by Milkshak3s** [comment]
This logging case is already covered in [sentry/src/sentry/auth/superuser.py](https://github.com/getsentry/sentry/blob/ea9e2e29653c5eb60741d68a1220657e24609eeb/src/sentry/auth/superuser.py#L328)

**Pull request closed without merging**